packages/editor/src/store/controls.js

```diff
@@ -21,10 +38,41 @@ export function dispatch( storeKey, actionName, ...args ) {
 	};
 }
 
-const controls = {
-	DISPATCH: createRegistryControl( ( registry ) => ( { storeKey, actionName, args } ) => {
-		return registry.dispatch( storeKey )[ actionName ]( ...args );
-	} ),
-};
+export default {
+	API_FETCH( { request } ) {
+		return triggerFetch( request );
+	},
+	SELECT: createRegistryControl(
+		( registry ) => ( { storeKey, selectorName, args } ) => {
+			return registry.select( storeKey )[ selectorName ]( ...args );
+		}
+	),
+	DISPATCH: createRegistryControl(
+		( registry ) => ( { storeKey, actionName, args } ) => {
+			return registry.dispatch( storeKey )[ actionName ]( ...args );
+		}
+	),
+	RESOLVE_SELECT: createRegistryControl(
+		( registry ) => ( { storeKey, selectorName, args } ) => {
+			return new Promise( ( resolve ) => {
+				const hasFinished = () => registry.select( 'core/data' )
+					.hasFinishedResolution( storeKey, selectorName, args );
+				const getResult = () => registry.select( storeKey )[ selectorName ]
+					.apply( null, args );
 
-export default controls;
+				// trigger the selector (to trigger the resolver)
+				const result = getResult();
```

Comment:
Minor: We could optimize this to avoid calling `getResult` until after we're certain it would be useful for the satisfied result of `hasFinished()`, i.e. call it inline with `return resolve( getResult() );` as done in the subscriber callback, or at least assigned as part of the `if` condition body.

Reply:
hmm I just copied this directly from the `resolveSelector` function in `effects/utils` assuming via the comment that the selector needed to be called to trigger the resolution (otherwise `hasFinished()` would never return true).  The comment block made sense to me here.  Still feel it's unnecessary?

Reply:
Ah, I neglected to consider the preceding comment. It feels awkward to rely on a side-effect from a function named as a getter, but I suppose that's the nature of resolvers. It should be fine as-is.